test/e2e/lua/dynamic_configuration.go

```diff
@@ -184,6 +184,68 @@ var _ = framework.IngressNginxDescribe("Dynamic Configuration", func() {
 			})
 		Expect(err).ToNot(HaveOccurred())
 	})
+
+	Context("when session affinity annotation is present", func() {
+		It("should use sticky sessions when ingress rules are configured", func() {
+			updateConfigmap("error-log-level", "info", f.KubeClientSet)
+
+			ingress, err := f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.Namespace.Name).Get("foo.com", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+	
+			ingress.ObjectMeta.Annotations = map[string]string{
+				"nginx.ingress.kubernetes.io/affinity": "cookie",
+			}
+			_, err = f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.Namespace.Name).Update(ingress)
+			Expect(err).ToNot(HaveOccurred())
+			time.Sleep(5 * time.Second)
+
+			host := "foo.com"
+			resp, _, errs := gorequest.New().
+				Get(f.NginxHTTPURL).
+				Set("Host", host).
+				End()
+			Expect(len(errs)).Should(BeNumerically("==", 0))
+			Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+
+			log, err := f.NginxLogs()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(log).ToNot(BeEmpty())
+			Expect(log).To(ContainSubstring("using lb_alg round robin with session affinity"))
+		})
+
+		It("should use sticky sessions when a default backend and no ingress rules configured", func() {
+			updateConfigmap("error-log-level", "info", f.KubeClientSet)
```

Comment:
The nginx configuration configmap is modified, but not reverted at the end of the test - meaning that `error-log-level` will stay `info` after the test as well.